### PR TITLE
35 remove reserved keywords clobbering

### LINKED ISSUE #35: global: refactor reserved keywords
- [x] `id` in the views/routes
- [x] `object` in (de)serializers
- [x] ... maybe more

### PR DESCRIPTION
**Pull request by fenekku**
NOTE: By introducing route arguments extraction, defining `*args, **kwargs` should not be necessary for Resource methods anymore I believe. I didn't change the signature though until you double-check.

closes #35

### CODE REVIEW

File: flask_resources/context.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def __init__(
         self.payload_mimetype = payload_mimetype  # Content-Type
         self.request_args = request_args
         self.request_content = data
+        self.route = {}
```
Comment (ppanero) [inline flask_resources/context.py:51]: minor: route_args? Is not putting the route per se, but its args no?

File: flask_resources/context.py [hunk 2]
```diff
@@ -72,3 +73,19 @@ def inner(*args, **kwargs):
             return f(*args, **kwargs)
 
     return inner
+
+
+def with_route(f):
```
Comment (ppanero) [inline flask_resources/context.py:78]: minor: if changed name above, here to `with_route_args`.

File: flask_resources/context.py [hunk 3]
```diff
@@ -72,3 +73,19 @@ def inner(*args, **kwargs):
...
+    """Wrapper to extract route arguments into resource context.
+
+    NOTE: This removes the need for *args, **kwargs passing in Resource methods
+    TODO: This should be temporary because the whole decorator approach should be
```
Comment (ppanero) [inline flask_resources/context.py:82]: Agree. Could you open an issue?

File: flask_resources/deserializers/json.py [hunk 4]
```diff
@@ -17,4 +17,4 @@ class JSONDeserializer(DeserializerMixin):
 
     def deserialize_data(self, data, *args, **kwargs):
         """Deserializes the input data into json."""
-        return json.loads(data)
+        return json.loads(data) if data else {}
```
Comment (ppanero) [inline flask_resources/deserializers/json.py:20]: Doubt: In which cases would the `else` clause happen?

File: flask_resources/context.py (hunk 1 above)
Comment (fenekku) [reply flask_resources/context.py:51]: Hmm I think `request_args` and `route_args` are confusing: both come from the request... And if we want to get away from the request aspect, we would want to remove `request` references too... Something tells me that every variables in the context object is an "argument" anyway (they are all inputs from the context of execution, which for now is a request). :thinking:  I've put a note in https://github.com/inveniosoftware/flask-resources/issues/2 to revisit naming and shelved it for now.

File: flask_resources/context.py (hunk 2 above)
Comment (fenekku) [reply flask_resources/context.py:78]: shelved as well.

File: flask_resources/context.py (hunk 3 above)
Comment (fenekku) [reply flask_resources/context.py:82]: Good. Wanted that double check. I created the issue:  https://github.com/inveniosoftware/flask-resources/issues/49

Comment (ppanero) [reply flask_resources/context.py:82]: IMHO the issue does not represent the comment 😅 There is no reference to "decorators". Mainly I do not fully agree with removing the `*args` and `**kwargs` so it feels like is the only solution there :(

Plus the `decorator` approach, needs to be reviewed in general (for all generic decorators).

File: flask_resources/context.py (hunk 1 above)
Comment (ppanero) [reply flask_resources/context.py:51]: Hmm you have a point... yes... this requires more thought.

File: flask_resources/deserializers/json.py (hunk 4 above)
Comment (fenekku) [reply flask_resources/deserializers/json.py:20]: For any request with no payload. For DELETE for instance. See the test I've added in the PR: https://github.com/inveniosoftware/flask-resources/pull/46/files#diff-c24f9f74a93cc9e07203da5e9e0522c3R70 . In this case, `json.loads(data)` would try to load an empty bytestream (empty `request.data`) and error out even though passing an empty payload is fine.

Comment (fenekku) [reply flask_resources/deserializers/json.py:20]: For requests wihout a payload. For instance, for DELETE call. See the test I've added in the PR: https://github.com/inveniosoftware/flask-resources/pull/46/files#diff-c24f9f74a93cc9e07203da5e9e0522c3R70 . Without this check, `json.loads(data)` tries to load an empty bytestream (`request.data` is empty) and errors out even though an empty payload is fine.

File: flask_resources/context.py (hunk 3 above)
Comment (fenekku) [reply flask_resources/context.py:82]: Ha my bad I read that wrong :sweat_smile: ! I made a point about it in https://github.com/inveniosoftware/flask-resources/issues/2

**Pull request merged**